### PR TITLE
Improve `prefer-includes` rule

### LINKED ISSUE #410: prefer-includes fix breaks utility library usage (eg. lodash)
`prefer-includes` rule autofix incorrectly applies to usages of utility libraries like lodash.

For example:

```js
let arr = [1, 2, 3];

if (_.indexOf(arr, 1) !== -1) {
    console.log("true");
}
```

Becomes
```js
let arr = [1, 2, 3];

if (_.includes(arr)) {
    console.log("true");
}
```

After the fix is applied, which is semantically different and leads to bugs.

### PR DESCRIPTION
**Pull request by fisker**
1. Ignore `indexOf` more than 2 arguments
2. Ignore `_.indexOf`, fixes #410
3. keep `Array/String.indexOf` method second argument [`fromIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf#Parameters)
4. strip `fromIndex` that is literal `0`

### CODE REVIEW

**Comment by sindresorhus** [comment]
> Ignore _.indexOf, fixes #410

Can you also ignore `lodash.indexOf` and `underscore.indexOf`?

**Comment by fisker** [comment]
> > Ignore _.indexOf, fixes #410
> 
> Can you also ignore `lodash.indexOf` and `underscore.indexOf`?

done

**Comment by fisker** [comment]
forgot test ...

**Comment by fisker** [comment]
@sindresorhus 

I just realize that `lodash` also has `includes`, so we can still replace `indexOf` with `includes`, but `underscore` don't has `includes` but `contains`

1, should I replace `lodash.indexOf` with `lodash.includes`?
2, should I replace `underscore.indexOf` with `underscore.contains`, rule name is `prefer-includes`?
3, still ignore `_.indexOf`?

**Comment by sindresorhus** [comment]
I would ignore them all. Yes, Lodash has `.includes()` today, but it might remove it in the future since it's already in the language now. Better to just leave it.

**Comment by fisker** [comment]
Then this PR is ready

File: rules/prefer-includes.js [hunk 1]
```diff
@@ -26,29 +32,40 @@ const create = context => ({
...
+		if (!isMethodNamed(left, 'indexOf')) {
+			return;
+		}
+
+		const target = left.callee.object;
 
-			if (right.type === 'UnaryExpression') {
-				const {argument} = right;
+		// Ignore {_,lodash,underscore}.indexOf
+		if (target.type === 'Identifier' && ['_', 'lodash', 'underscore'].includes(target.name)) {
```
Comment (MrHen) [inline rules/prefer-includes.js]: This shouldn't be hardcoded here. Either (a) convert it to a blacklist configuration property with these values as a default or (b) move the values to a constant at the top of the file.

Comment (fisker) [reply rules/prefer-includes.js]: will fix it tomorrow with (b) 

I think it should be whitelist not blacklist, right?

Comment (MrHen) [reply rules/prefer-includes.js]: The phrase "ignore" is less ambiguous. Maybe something like `ignoreWords`.

Comment (fisker) [reply rules/prefer-includes.js]: Good idea, going to get up and fix it now

**Comment by sindresorhus** [comment]
Nice work on improving this rule 👌

**Pull request merged**